Server/src/main/resources/static/js/editor.js

```diff
@@ -260,136 +260,96 @@ function getOperandType(text, result) {
+    const oneArgInstructions = "push|mul|pop|div|neg|call|jnz|jg|jl|jge|jle|hwi|hwq|jz|js|jns|ret|jmp|not|jc|jnc|jo|jno|inc|dec|ja|jna|seta|setnbe|setae|setnb|setnc|setbe|setna|setb|setc|setnae|sete|setz|setne|setnz|setg|setnle|setge|setnl|setle|setng|setl|setnge|seto|setno|sets|setns".split("|");
+    const zeroArgInstructions = "brk|into|iret|nop|popf|pushf|ret"
+
+    const nonImmediateSet = "seta|setnbe|setae|setnb|setnc|setbe|setna|setb|setc|setnae|sete|setz|setne|setnz|setg|setnle|setge|setnl|setle|setng|setl|setnge|seto|setno|sets|setns".split("|")
+
+    //there are some duplicates here but it doesn't matter since we are just trying to find membership
+    const concatenatedInstructions = twoArgInstructions + oneArgInstructions + zeroArgInstructions
+
+    const instructionsByArgCount = [zeroArgInstructions, oneArgInstructions, twoArgInstructions]
+
+    let tokens = getTokens(passedCurrentLine)   // get the actual tokens
+    tokens.shift() //removing the first element from array, since that's the mnemonic and we already have that
+    const tokenCount = tokens.length()
+    
+    //select which set of instructions are admissable for this number of arguments
+    correspondingInstructionSet = instructionsByArgCount[tokenCount]
+    
+    if (correspondingInstructionSet.includes(passedMnemonic)) {
+        //this means that for the number of arguments is valid for this opcode, and that the code itself is a recognized code
+
+        //handle cases for forbidden use of immediate values in one and two token cases
+        if (tokenCount === 2 && getOperandType(tokens[0] === OPERAND_IMM)) {
+            //fail because destination operand can't be immediate value
+            passedResult.annotations.push(resultObjectBuilder("Destination operand for: " + passedMnemonic + " can't be an immediate value"))
+            return
+        }
+        else if (tokenCount === 1 && nonImmediateSet.includes(passedMnemonic) && getOperandType(tokens[0] === OPERAND_IMM)) {
+            passedResult.annotations.push(resultObjectBuilder("Invalid operand type for: " + passedMnemonic + " operand must not be immediate value"))
+            return
+        }
+
+        for (string in tokens) {
+            if (getOperandType(tokens[string]) === OPERAND_INVALID){
+                //fail for invalid operand type
+                passedResult.annotations.push(resultObjectBuilder("Invalid operand type: " + passedMnemonic))
+                return
+            }
+        }
+
+        //if we make it here then the operation type is a valid operation, the opcode is recognized, and the number of arguments is correct so why did we fail?
+    }
+    else if (concatenatedInstructions.includes(passedMnemonic)) {
+            //wrong number of arguments for instruction, but it is SOMEWHERE in this list so it's got to be one of these 3 values
+            
+            let expectedNumber = 0
+            if ( twoArgInstructions.includes(passedMnemonic) ) { expectedNumber=2}
```

Comment:
here there should be no spaces after the parentheses and there should be a newline after the brackets like
```
if (...) {
    ...
} else if (...) {
    ....
}
```

Reply:
I will totally get to this